File: garage-microservice/src/main/java/com/tunidesign/garagemicroservice/model/Garage.java
package com.tunidesign.garagemicroservice.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "garage")
public class Garage {
    @Transient
    public static final String SEQUENCE_NAME = "garage_sequence";

    @Id
    private Long id;
    private String name;
    private String email;
    private Integer phone;
    private String countryCode;
    private String dialCode;
    private String adresse;
    private String garageType;
    private Long garageOwner;
    private int capacite;
}

File: Front-End/src/app/models/garage/garage.ts
export class Garage {
    name!: string;
    capacite!: number;
    garageOwner!: number;
    email!: string;
    phone!: number;
    countryCode!: string;
    dialCode!: string;

    adresse!: string;
    garageType!: string;

    public constructor();
    public constructor(name: string, capacity: number, garageOwner: number, email: string, 
        phone: number, adresse: string);
    public constructor(...myarray: any[]){
        this.name = myarray[0];
        this.capacite = myarray[1];
        this.email = myarray[2];
        this.garageOwner = myarray[3];
        this.email = myarray[4];
        this.phone = myarray[5];
        this.adresse = myarray[6];
    }
}
